### PR TITLE
[Quantization] Enable FP8 weight storage for Qwen Image VAE and text encoder

### PR DESCRIPTION
**Pull request by lishunyang12**
## Summary
- Add FP8 weight-only storage for Linear/Conv2d/Conv3d layers in the Qwen Image **VAE** and **text encoder** (`Qwen2_5_VLForConditionalGeneration`)
- Weights are stored in `float8_e4m3fn` with per-tensor scales and dequantized to BF16 before each forward pass — saving ~50% memory for these components with no accuracy loss
- Applied to all three Qwen Image pipelines: generation, edit, and edit-plus
- Fixed `load_weights()` to mark VAE and text_encoder parameters as loaded (they use `from_pretrained()`, not the weight pipeline)

## Approach
New utility `apply_fp8_weight_storage()` in `vllm_omni/diffusion/models/utils.py`:
- Walks all `nn.Linear`, `nn.Conv2d`, and `nn.Conv3d` modules (including subclasses like `QwenImageCausalConv3d`)
- Computes per-tensor scale, converts weight to FP8, stores FP8 weight + scale as buffers
- Registers forward pre-hook (dequantize to BF16) and post-hook (re-quantize to FP8)
- Peak memory per layer: FP8 + BF16 weight only during that layer's forward; all other layers stay in FP8

This covers all three non-DiT components when `--quantization fp8` is passed:
| Component | Quantization method |
|---|---|
| DiT (transformer) | vLLM FP8 linear layers (PR #1338) |
| Text encoder | FP8 weight storage (this PR) |
| VAE | FP8 weight storage (this PR) |

## Test plan
- [ ] Run Qwen Image text-to-image with `--quantization fp8` and verify output quality matches BF16 baseline
- [ ] Verify memory reduction for VAE and text encoder vs without `--quantization fp8`
- [ ] Run pre-commit checks (ruff lint + format)

## Related
Follow-up to #1338 (FP8 online quantization for DiT transformer)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
for the weights, i remember there are two versions for both qwen-image and qwen-image-edit.

**Comment by hsliuustc0106** [comment]
@vllm-omni-reviewer

**Comment by hsliuustc0106** [comment]
@vllm-omni-reviewer

File: vllm_omni/diffusion/models/utils.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        # Quantize weight to FP8
+        fp8_weight = (weight / scale).clamp(min=-_FP8_E4M3_MAX, max=_FP8_E4M3_MAX).to(torch.float8_e4m3fn)
+
+        # Store FP8 weight and metadata as buffers (not parameters)
+        module.register_buffer("_fp8_weight", fp8_weight)
+        module.register_buffer("_fp8_scale", scale.to(torch.float32))
+        module._fp8_compute_dtype = compute_dtype
+
+        # Replace the parameter data with FP8 to save memory
+        module.weight.data = fp8_weight
```
[the remark this answers is not in the document]
Comment (lishunyang12) [reply vllm_omni/diffusion/models/utils.py]: Fixed in the latest push. `module.weight.data` now stays at compute dtype (BF16/FP16). The FP8 representation is stored in the `_fp8_weight` buffer and only dequantized into `module.weight.data` inside the pre-hook. The post-hook writes back `mod._fp8_weight.to(mod._fp8_compute_dtype)` instead of raw FP8, so `model.dtype` stays correct at all times.

Also added an idempotency guard (`hasattr(module, "_fp8_weight")`) to prevent double-application.

File: vllm_omni/diffusion/models/utils.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        def _pre_hook(mod, args):
+            # Dequantize: restore BF16/FP16 weight for conv computation
+            mod.weight.data = mod._fp8_weight.to(mod._fp8_compute_dtype) * mod._fp8_scale
```
[the remark this answers is not in the document]
Comment (lishunyang12) [reply vllm_omni/diffusion/models/utils.py]: Fixed. The dequantized result is now explicitly cast back to `compute_dtype` after multiplication with the float32 scale:

```python
# Before
mod.weight.data = mod._fp8_weight.to(mod._fp8_compute_dtype) * mod._fp8_scale

# After
mod.weight.data = (
    mod._fp8_weight.to(mod._fp8_compute_dtype) * mod._fp8_scale
).to(mod._fp8_compute_dtype)
```

**Comment by lishunyang12** [comment]
> for the weights, i remember there are two versions for both qwen-image and qwen-image-edit.

@hsliuustc0106 Could you clarify which two versions you mean? Is this about different checkpoint formats (e.g., safetensors vs pytorch_model.bin), different model sizes, or different training stages? I want to make sure we test with both.

**Comment by hsliuustc0106** [comment]
Hi @lishunyang12 👋

This PR for FP8 weight storage hasn't been updated for 15 days. Just checking if this is still in progress.

Thanks!

**Comment by lishunyang12** [comment]
Merged into #1412 — combining Wan 2.2 FP8 + Qwen Image VAE/encoder FP8 into a single PR.
